lib_tflite_micro/api/xcore_shared_config.h

```diff
@@ -9,8 +9,19 @@ namespace shared_config {
 constexpr char xcoreMetadataName[] = "xcoreSharedConfig";
 
 struct xcore_metadata {
+  // Versions of libraries used to build the model
+  int lib_nn_major_version;
```

Comment:
I don't mind which way you go with this but wouldn't declaring a struct/class be a nicer way of doing this?

Reply:
i didn't quite understand, it is a struct at the moment. Did you mean another struct for each library? We are sharing this struct between the compiler and runtime and want to keep it simple. Let me know if we can tidy it up another way.